src/sidebar/services/threads.js

```diff
@@ -0,0 +1,18 @@
+// @ngInject
+export default function threadsService(store) {
+  /**
+   * Make this thread and all of its children "visible". This has the effect of
+   * "unhiding" a thread which is currently hidden by an applied search filter
+   * (as well as its child threads).
+   */
+  function forceVisible(thread) {
```

Comment:
This logic reproduces what I believe several functions are trying to accomplish in `annotation-thread.js`. Start here:

https://github.com/hypothesis/client/blob/b4953ce60964ec71c10a703f6ceef2d167dad2e3/src/sidebar/components/annotation-thread.js#L60-L64

At the end of the day, this logic seems to "work." The right things are being expanded and collapsed. In this current branch, you can see both the old `annotation-thread` and the new `Thread` alongside each other, so you can test to see if any of the expand/collapse behavior is different between the two.

Reply:
This worked as I expected from a user's point of view, so the new code here is fine with me.

Reply:
That's good news because the new variant is much simpler! Why use four functions when you can use one? :)